### PR TITLE
Migrate GeomDetLess and GluedGeomDet to Geometry/CommonDetUnit

### PR DESCRIPTION
**Pull request by ianna**
- Migrate GeomDetLess and GluedGeomDet to Geometry/CommonDetUnit

@smuzaffar and @Dr15Jones - as discussed at yesterday's meeting

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by smuzaffar** [comment]
cmsbuild, please test

File: Geometry/CommonDetUnit/interface/GeomDetLess.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#ifndef TrackerGeometryBuilder_GeomDetLess_H
+#define TrackerGeometryBuilder_GeomDetLess_H
+
+#include "DataFormats/TrajectorySeed/interface/PropagationDirection.h"
```
Comment (Dr15Jones) [inline Geometry/CommonDetUnit/interface/GeomDetLess.h]: The BuildFile.xml for Geometry/CommonDetUnit should also be changed to depend upon DataFormats/TrajectorySeed.

File: Geometry/CommonDetUnit/interface/GluedGeomDet.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#ifndef Geometry_TrackerGeometryBuilder_GluedGeomDet_H
+#define Geometry_TrackerGeometryBuilder_GluedGeomDet_H
+
+#include "Geometry/CommonDetUnit/interface/TrackerGeomDet.h"
+#include "DataFormats/DetId/interface/DetId.h"
```
Comment (Dr15Jones) [inline Geometry/CommonDetUnit/interface/GluedGeomDet.h]: The BuildFile.xml file for Geometry/CommonDetUnit should also be changed to depend upon DataFormats/DetId.

File: Geometry/CommonDetUnit/interface/GeomDetLess.h (hunk 1 above)
Comment (ianna) [inline Geometry/CommonDetUnit/interface/GeomDetLess.h]: This include defines an enumerator. However, adding here a direct dependency on DataFormats/TrajectorySeed introduces circular dependency:

```
tmp/slc6_amd64_gcc530/src/Geometry/CommonDetUnit/src/GeometryCommonDetUnit/libGeometryCommonDetUnit.so <- tmp/slc6_amd64_gcc530/cache/prod/libGeometryCommonDetUnit
```

File: Geometry/CommonDetUnit/interface/GluedGeomDet.h (hunk 2 above)
Comment (ianna) [inline Geometry/CommonDetUnit/interface/GluedGeomDet.h]: done.

**Comment by ianna** [comment]
please test

File: Geometry/CommonDetUnit/interface/GeomDetLess.h (hunk 1 above)
Comment (Dr15Jones) [inline Geometry/CommonDetUnit/interface/GeomDetLess.h]: Then it seems to me the implication is this header actually belongs in DataFormats/TrajectorySeed.

**Comment by dmitrijus** [comment]
+1

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by lveldere** [comment]
+1

File: Geometry/CommonDetUnit/interface/GluedGeomDet.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#ifndef Geometry_TrackerGeometryBuilder_GluedGeomDet_H
+#define Geometry_TrackerGeometryBuilder_GluedGeomDet_H
```
Comment (slava77) [inline Geometry/CommonDetUnit/interface/GluedGeomDet.h]: the include guard is wrong

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/14455	6dd345e
- changes are as described
- jenkins tests pass and comparisons show no differences

The issue in  Geometry/CommonDetUnit/interface/GluedGeomDet.h  include guard is rather minor and is not in reco area.

File: Geometry/CommonDetUnit/interface/GluedGeomDet.h (hunk 3 above)
Comment (ianna) [inline Geometry/CommonDetUnit/interface/GluedGeomDet.h]: @slava77  - Thanks for pointing it out. I'll fix it in the next PR along with cleaning up spurious dependencies on Geometry package.

**Pull request merged**